ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java

```diff
@@ -722,11 +723,26 @@ public void commandSent(GcodeCommand command) {
         dispatchCommandSent(command);
     }
 
+    @Override
+    public void communicatorPaused() {
+        dispatchConsoleMessage(MessageType.INFO, "**** The communicator has been paused ****\n");
+        try {
+            // FIXME the dispatchStateChange has logic for resuming send operations, this should have it's own handler.
```

Comment:
This could use a comment that it is specifically circumventing the `setCurrentState`/`handleAllStateChangeEvents` in case of an error (also see the other comment about renaming this callback).